activerecord/lib/active_record/associations/builder/belongs_to.rb

```diff
@@ -17,6 +17,12 @@ def self.valid_dependent_options
       [:destroy, :delete, :destroy_async]
     end
 
+    def self.validate_reflection!(model, reflection)
+      if reflection.belongs_to? && model.ignored_columns.include?(reflection.foreign_key.to_s)
```

Comment:
Could `reflection.belongs_to?` be omitted? We are in a `belongs_to` reflection builder so `reflection` being `belongs_to` should be implied. Unless there is a corner case I'm missing 

```suggestion
      if model.ignored_columns.include?(reflection.foreign_key.to_s)
```

Reply:
@nvasilevski  thank you so much for pointing out the redundant belongs_to? check. I just pushed the refactoring. Let me know if that is okey.

Regarding when foreign key does not exists: I think we can consider a new pull request. what do you think? I think we can keep this pull request only for the given context. Let me know your feedback .